tensorstore/kvstore/s3/BUILD

```diff
@@ -0,0 +1,215 @@
+load("//bazel:tensorstore.bzl", "tensorstore_cc_library", "tensorstore_cc_test")
+
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
```

Comment:
You can make your extra logging build-time defines (TENSORSTORE_INTERNAL_S3_LOG_REQUESTS...)
by adding something like the following to the BUILD file.

```
# To enable debug checks, specify:
# blaze build --//tensorstore/kvstore/s3:debug
bool_flag(
  name = "debug",
  build_setting_default = False,
)

config_setting(
    name = "debug_setting",
    flag_values = {
        ":debug": "True",
    },
    visibility = ["//visibility:private"],
)

DEBUG_LOCAL_DEFINES = select({
    "//conditions:default": [],
    ":debug_setting": [
        "TENSORSTORE_INTERNAL_S3_LOG_REQUESTS",
        "TENSORSTORE_INTERNAL_S3_LOG_RESPONSES",
      ],
})
```

Then in each library, use `local_defines = DEBUG_LOCAL_DEFINES`

You can also add a `.bazelrc` line to keep them enabled like this:

`build  --//tensorstore/kvstore/s3:debug`